op/framework.py

```diff
@@ -374,14 +372,25 @@ def __init__(self, data_path, charm_dir, meta, model):
         self.charm_dir = charm_dir
         self.meta = meta
         self.model = model
-        self._event_count = 0
         self._observers = []      # [(observer_path, method_name, parent_path, event_key)]
         self._observer = weakref.WeakValueDictionary()       # {observer_path: observer}
         self._type_registry = {}  # {(parent_path, kind): cls}
         self._type_known = set()  # {cls}
 
         self._storage = SQLiteStorage(data_path)
 
+        # Note: we can't use the higher-level StoredData because it relies on events.
```

Comment:
s/Note: // (every comment is a "please note this")

Also, this is about StoredState, not StoredStateData.. the latter is what's being used.